src/core/abstracts/SablierLockup.sol

```diff
@@ -379,6 +379,12 @@ abstract contract SablierLockup is
             revert Errors.SablierLockup_WithdrawAmountZero(streamId);
         }
 
+        // Check: the withdraw amount is not greater than the withdrawable amount.
+        uint128 withdrawableAmount = _withdrawableAmountOf(streamId);
+        if (amount > withdrawableAmount) {
+            revert Errors.SablierLockup_Overdraw(streamId, amount, withdrawableAmount);
```

Comment:
>  i think it’s best to exit as quickly and cheaply as possible

I agree with your point. The reason I moved it above `if (to != recipient && !_isCallerStreamRecipientOrApproved(streamId))` is so that it can follow the tree structure. There, we have `withdraw amount overdraws` branch before `withdrawal address` branches.

But it does not matter. Tree branches are not supposed to follow the same order as the logic. I can revert it back.

> if you would like to group the amount checks we could move the if (amount == 0) check lower right before withdrawable

Thats what it is right now. No? amount check is before withdrawable call.

Reply:
> Thats what it is right now. No? amount check is before withdrawable call.

this is what i meant: 76b6f5055956722b7c01a8169642320e923f2f0b

Reply:
I see. Looks great to me.